packages/trpc/server/routers/viewer/eventTypes/update.handler.ts

```diff
@@ -240,49 +240,31 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
     }
   }
 
-  // Find a way to handle this 3 payment app better
-  if (input.metadata?.apps?.stripe?.price) {
-    data.price = input.metadata?.apps?.stripe?.price;
-    const paymentCredential = await ctx.prisma.credential.findFirst({
-      where: {
-        userId: ctx.user.id,
-        appId: {
-          equals: "stripe",
-        },
-      },
-      select: {
-        appId: true,
-        key: true,
-      },
-    });
-
-    if (paymentCredential?.appId === "stripe") {
-      const { default_currency } = stripeDataSchema.parse(paymentCredential.key);
-      data.currency = default_currency;
-    }
+  /**
+   * Since you can have multiple payment apps we will honor the first one to save in eventType
+   * but the real detail will be inside app metadata, so with this you can have different prices in different apps
+   * So the price and currency inside eventType will be deprecated soon or just keep as reference.
+   */
+  if (
```

Comment:
This has exposed a bug. I only have Alby installed (stripe is not installed at https://endless-close-ocelot.ngrok-free.app/apps/stripe and is not listed in my apps on the event type), but `input.metadata.apps` also contains stripe and therefore the stripe price and currency are picked since they are checked first. @alannnc can you help here?

I think how this works is a bit broken - either it should be enforced that there can only be one payment app per event type (probably best for now?) or add proper support for accepting multiple payment methods and do not store a `price` field on the event type.

```
"metadata": {
    "apps": {
      "alby": {
        "enabled": true,
        "price": 1,
        "currency": "BTC"
      },
      "giphy": {
        "enabled": false,
        "thankYouPage": ""
      },
      "stripe": {
        "enabled": true,
        "price": 1,
        "currency": "usd"
      }
    }
  },
```

![image](https://github.com/calcom/cal.com/assets/33993199/95b4eab0-73a2-4dc8-b62f-ea2e618e5b9e)

Reply:
Yeah I ended in the same path, looks like legacy bug since it also saves giphy which shouldn't be the case. My quick fix was changing order of paymet apps so it would take alby | paypal first before stripe.

Reply:
@alannnc so you already fixed it locally? can you push the fix?

Reply:
Should be fixed now.